Kuroba/app/src/main/java/com/github/adamantcheese/chan/features/gesture_editor/Android10GesturesExclusionZonesHolder.kt

```diff
@@ -0,0 +1,236 @@
+package com.github.adamantcheese.chan.features.gesture_editor
+
+import android.graphics.Rect
+import com.github.adamantcheese.chan.core.settings.ChanSettings
+import com.github.adamantcheese.chan.core.settings.ChanSettings.EMPTY_JSON
+import com.github.adamantcheese.chan.utils.AndroidUtils.isAndroid10
+import com.github.adamantcheese.chan.utils.Logger
+import com.google.gson.Gson
+
+class Android10GesturesExclusionZonesHolder(
+        private val gson: Gson,
+        private val minScreenSize: Int,
+        private val maxScreenSize: Int
+) {
+    private val exclusionZones: MutableMap<Int, MutableSet<ExclusionZone>> = loadZones()
+
+    private fun loadZones(): MutableMap<Int, MutableSet<ExclusionZone>> {
+        val zones = mutableMapOf<Int, MutableSet<ExclusionZone>>()
+
+        val json = ChanSettings.androidTenGestureZones.get()
+        if (json.isEmpty()) {
+            Logger.d(TAG, "Json setting string is empty, reset.")
+            ChanSettings.androidTenGestureZones.set(EMPTY_JSON)
+            return zones
+        }
+
+        if (json == EMPTY_JSON) {
```

Comment:
You can collapse this check into the one above.

Reply:
Done.